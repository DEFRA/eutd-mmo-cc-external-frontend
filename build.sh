#/bin/bash
curl -O https://nodejs.org/dist/v12.18.4/node-v12.18.4-linux-x64.tar.xz
tar -xf node-v12.18.4-linux-x64.tar.xz --directory /tmp
export PATH="/tmp/node-v12.18.4-linux-x64/bin:$PATH"
rm node-v12.18.4-linux-x64.tar.xz
npm ci
npm run test-vstack
rm -rf /tmp/node-v12.18.4-linux-x64