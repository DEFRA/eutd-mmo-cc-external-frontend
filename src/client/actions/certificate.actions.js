import { isError, showFullPageError } from './index';

export const GET_SUMMARY_DOCUMENT = 'GET_SUMMARY_DOCUMENT';
export const SHOW_SUMMARY_DOCUMENT_ERRORS = 'SHOW_SUMMARY_DOCUMENT_ERRORS';
export const GET_SUMMARY_UNAUTHORISED = 'GET_SUMMARY_DOCUMENT_UNAUTHORISED';

export function showSummaryCertificateErrors(data) {
  return { type: SHOW_SUMMARY_DOCUMENT_ERRORS, payload: { validationErrors: data } };
}

export const getSummaryCertificate = (journey, documentNumber = undefined) => async (dispatch, getState, { orchestrationApi }) => {
  const config = (documentNumber)
    ? { headers: { documentnumber: documentNumber } }
    : {};

  await orchestrationApi.get(`/certificate/${journey}`, config)
    .then(
      res => {
        dispatch({ type: GET_SUMMARY_DOCUMENT, payload: res.data });
      },
      error => {
        if (isError(error)) {
          dispatch(showFullPageError());
        } else {
          dispatch(unauthorised());
        }
      });

    function unauthorised() { return { type: GET_SUMMARY_UNAUTHORISED }; }
};