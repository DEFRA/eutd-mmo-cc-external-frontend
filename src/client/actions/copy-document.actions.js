import { showFullPageError, apiCallFailed, isError, CLEAR_ERRORS } from './index';
export const CONFIRM_COPY_DOCUMENT            = 'confirm_copy_document';
export const UNAUTHORISED_COPY_DOCUMENT       = 'unauthorised_copy_document';
export const CLEAR_COPY_DOCUMENT              = 'clear_copy_document';
export const REMOVE_CONFIRM_COPY_CERTIFICATE  = 'remove_confirm_copy_certificate';
export const ADD_ERROR_COPY_CERTIFICATE       = 'add_error_copy_certificate';
export const REMOVE_ERROR_COPY_CERTIFICATE    = 'remove_error_copy_certificate';

export const submitCopyCertificate = (payload) => async (dispatch, _getState, {orchestrationApi}) => {
  const {documentNumber, copyDocumentAcknowledged, voidOriginal, excludeLandings, journey} = payload;
  const config = {headers: {documentnumber: documentNumber}};

  try {
    let ipAddress;
    if (voidOriginal) {
      const ip = await orchestrationApi.get('/client-ip');
      ipAddress = ip.data;
    }

    return await orchestrationApi.post('/confirm-copy-certificate', { copyDocumentAcknowledged, voidOriginal, excludeLandings, journey, ipAddress }, config)
    .then(res => {
      dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: {
        documentNumber: res.data.documentNumber,
        copyDocumentAcknowledged: res.data.copyDocumentAcknowledged,
        voidDocumentConfirm: res.data.voidOriginal
      }});
      return res.data.newDocumentNumber;
    });
  } catch (e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch({ type: UNAUTHORISED_COPY_DOCUMENT }); return;
    } else {
      dispatch(apiCallFailed(e.response));
    }
  }
};

export const checkCopyCertificate = (documentNumber) => async (dispatch, _getState, {orchestrationApi}) => {
  const config = {
    headers: {
      documentnumber: documentNumber
    }
  };

  try {
    const res = await orchestrationApi.get('/check-copy-certificate', config);
    const { canCopy } = res.data;
    if (!canCopy) {
      dispatch(unauthorised());
    }
  } catch(e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch(unauthorised());
      throw new Error(`Unauthorised access ${e.response.status}`);
    } else {
      dispatch(apiCallFailed(e.response));
      throw new Error(`An error has occurred ${e.response.status} (${e.response.statusText})`);
    }
  }

  function unauthorised() { return { type: UNAUTHORISED_COPY_DOCUMENT }; }
};

export const addCopyDocument = (data) => (dispatch) => dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: data });
export const clearErrors = () => (dispatch) =>  dispatch({ type: CLEAR_ERRORS });
export const clearCopyDocument = () => (dispatch) => dispatch({ type: CLEAR_COPY_DOCUMENT });
export const unauthorisedCopyDocument = () => (dispatch) => dispatch({ type: UNAUTHORISED_COPY_DOCUMENT });