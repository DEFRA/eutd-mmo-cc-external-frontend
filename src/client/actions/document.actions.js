// actions
import {isError, showFullPageError} from './index';

export const GOT_ALL_DOCUMENTS          = 'GOT_ALL_DOCUMENTS';
export const GET_ALL_DOCUMENTS_FAILED   = 'GET_ALL_DOCUMENTS_FAILED';
export const GET_COMPLETED_DOCUMENT = 'GET_COMPLETED_DOCUMENT';
export const GET_PENDING_DOCUMENT = 'GET_PENDING_DOCUMENT';


// action creators
export const getAllDocuments = (type, month, year) => async (dispatch, getState, { orchestrationApi }) => {
  try {
    const res = await orchestrationApi.get(`/documents/${year}/${month}?type=${type}`);
    return dispatch({ type: GOT_ALL_DOCUMENTS, payload: res.data });

  } catch(e) {
    if( isError(e) ) dispatch(showFullPageError());
    //throw new Error(e.response);
  }
};

export const getCompletedDocument = (documentNumber) => async (dispatch, getState, { orchestrationApi }) => {
  try {
    const res = await orchestrationApi.get(`/document/${documentNumber}`);
    if (res.data && res.data.documentStatus === 'COMPLETE') {
      return dispatch({ type: GET_COMPLETED_DOCUMENT, payload: res.data });
    }
    return dispatch({ type: GET_COMPLETED_DOCUMENT, payload: null });
  }
  catch (e) {
    return dispatch({ type: GET_COMPLETED_DOCUMENT, payload: null });
  }
};

export const getPendingDocument = (documentNumber) => async (dispatch, getState, { orchestrationApi }) => {
  try {
    const res = await orchestrationApi.get(`/document/${documentNumber}`);
    if (res.data && res.data.documentStatus === 'PENDING') {
      return dispatch({ type: GET_PENDING_DOCUMENT, payload: res.data });
    }
    return dispatch({ type: GET_PENDING_DOCUMENT, payload: null });
  }
  catch (e) {
    return dispatch({ type: GET_PENDING_DOCUMENT, payload: null });
  }
};