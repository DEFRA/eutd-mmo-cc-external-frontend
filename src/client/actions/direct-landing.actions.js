import { beginApiCall, apiCallFailed, isForbidden, showFullPageError, CLEAR_ERRORS, isSilverLineError } from './index';
import { exportPayloadActionTypes } from '../actions';


export const GET_DIRECT_LANDING_PRODUCTS = 'get_direct_landing_products';
export const CLEAR_DIRECT_LANDING = 'clear_direct_landing';
export const DIRECT_LANDING_PRODUCTS_FAILED = 'direct_landing_products_failed';
export const POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'post_direct_landing_products_unauthorised';
export const GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'get_direct_landing_products_unauthorised';
export const POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR = 'post_direct_landing_prodcts_waf_error';

export const getDirectLandings =
  (documentNumber) =>
  async (dispatch, _getState, { orchestrationApi }) => {
    const config = { headers: { documentnumber: documentNumber } };

    try {
      dispatch(beginApiCall());

      const res = await orchestrationApi.get(
        '/export-certificates/export-payload/direct-landings',
        config
      );

      await dispatch({ type: GET_DIRECT_LANDING_PRODUCTS, payload: res.data });
    } catch (e) {
      if (isForbidden(e)) {
        dispatch({ type: GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
      } else {
        dispatch(apiCallFailed(e.response));
        throw new Error(
          `An error has occurred ${e.response.status} (${e.response.statusText})`
        );
      }
    }
  };

export const clearDirectLanding = () => ({ type: CLEAR_DIRECT_LANDING });

export const upsertDirectLanding = (landing, documentNumber = undefined) => (dispatch, getState, { orchestrationApi } ) => {
  const config = (documentNumber)
    ? {
        headers: {
          documentnumber: documentNumber
        }
    }
    : {};

  return orchestrationApi.post('/export-certificates/direct-landing/validate', landing, config)
    .then(
      res => {
        dispatch(landingUpserted(res.data));
        dispatch(clearErrors());
      },
      error => {
        if (error.response.status === 400) {
          dispatch(landingUpserted(error.response.data));
          throw new Error();
        } else if (isSilverLineError(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR, supportID: error.response.data.match(/\d+/)[0] });
        } else if (isForbidden(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
        } else {
          dispatch(showFullPageError());
        }
      });

  function landingUpserted(exportPayload) { return { type: exportPayloadActionTypes.LANDING_UPSERTED, exportPayload, errors: exportPayload.errors }; }
  function clearErrors() { return { type: CLEAR_ERRORS }; }
};
