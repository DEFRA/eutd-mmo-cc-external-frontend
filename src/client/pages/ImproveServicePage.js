import React, { Component } from 'react';
import { connect } from 'react-redux';
import i18n from '../../../src/i18n';
import { getAllUserAttributes } from '../actions/userAttributes.actions';
import PageTemplateWrapper from '../components/PageTemplateWrapper';
import ImproveServiceWelshPage from './ImproveServiceWelshPage';
import ImproveServiceEnglishPage from './ImproveServiceEnglishPage';

function getService(backRoute) {
  if( backRoute.indexOf( '/create-processing-statement' ) !== -1) return {title: i18n.t('psCommonTitle'), href: '/create-processing-statement/processing-statements' };
  if( backRoute.indexOf( '/create-storage-document' ) !== -1) return {title: i18n.t('sdCommonTitle'), href: '/create-storage-document/storage-documents' };
  if( backRoute.indexOf( '/create-catch-certificate' ) !== -1) return {title: i18n.t('ccCommonTitle'), href: '/create-catch-certificate/catch-certificates' };
  return {title: '', href: '#'};
}
class ImproveServicePage extends Component {

  state = {
    backRoute: '#'
  };

  componentDidMount() {
    const backRoute = window.location.search.replace('?', '');
    this.setState( {backRoute});
    const service = getService(backRoute);
    const titleNode = document.getElementsByClassName('govuk-header__link--service-name');
    if( titleNode && titleNode[0] ) {
      titleNode[0].text = service.title.replace(' - GOV.UK', '');
      titleNode[0].href = service.href;
    }
  }

  getSelectedLanguage = (userAttributes) => {
    const userLanguageAttribute = userAttributes.find(
      (userAttribute) => userAttribute.name === 'language'
    );
    if (userLanguageAttribute) {
      return userLanguageAttribute.value;
    }
  };

  render() {
    const { backRoute } = this.state;
    const {feedbackUrl, userAttributes } = this.props
    const userSelectedLanguage = this.getSelectedLanguage(userAttributes);

    let title = getService(backRoute).title;
    if( !title ) title = 'GOV.UK';

    const listFont = {
      'fontSize': '19px'
    };

    return userSelectedLanguage === 'cy_UK' ? (
      <ImproveServiceWelshPage
        feedbackUrl={feedbackUrl}
        backRoute={backRoute}
        listFont={listFont}
        goBack={() => this.props.history.goBack()}
        title={title}
      />
    ) : (
      <ImproveServiceEnglishPage
        feedbackUrl={feedbackUrl}
        backRoute={backRoute}
        listFont={listFont}
        goBack={() => this.props.history.goBack()}
        title={title}
      />
    );
  }
}

function mapStateToProps(state) {
  return {
    feedbackUrl: state.config.feedbackUrl,
    userAttributes: state.userAttributes,
  };
}

function loadData(store) {
  return Promise.all([store.dispatch(getAllUserAttributes())]);
}

export const component = connect(mapStateToProps, {
  getAllUserAttributes,
})(ImproveServicePage);

export default {
  loadData,
  component: PageTemplateWrapper(component),
};
