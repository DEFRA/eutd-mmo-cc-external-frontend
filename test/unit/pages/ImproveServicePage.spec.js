import { mount } from 'enzyme/build';
import { Provider } from 'react-redux';
import * as React from 'react';
import { component as ImproveServicePage } from '../../../src/client/pages/ImproveServicePage';
import ImproveServicePageComponent from '../../../src/client/pages/ImproveServicePage';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';
import { getAllUserAttributes } from '../../../src/client/actions/userAttributes.actions';

jest.mock('../../../src/client/actions/userAttributes.actions');

const mockStore = configureStore([
  thunk.withExtraArgument({
    orchestrationApi: {
      get: () => {
        return new Promise((res) => {
          res({});
        });
      },
    },
  }),
]);

let wrapper;
const mockGoBack = jest.fn();

const getWrapper = (name, language) => {
  const store = mockStore({
    config: {
      feedbackUrl: '/some-feedback-url',
    },
    userAttributes: [
      {
        name: name,
        value: language,
        modifiedAt: '2022-03-15T21:47:10.755Z',
      },
    ],
  });
  const props = {
    route: {
      title: '',
      previousUri: '/create-catch-certificate/catch-certificates',
    },
    history: {
      goBack: mockGoBack,
    },
  };

  wrapper = mount(
    <Provider store={store}>
      <ImproveServicePage {...props} />
    </Provider>
  );

  return wrapper;
};

describe('Improve service page in English', () => {
  beforeEach(() => {
    jest.spyOn(document, 'getElementsByClassName').mockReturnValue([
      {
        title: '',
        href: '',
      },
    ]);

    wrapper = getWrapper('language', 'en_UK');
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should load successfully', () => {
    expect(wrapper).toBeDefined();
  });

  it('should handle clicking back link', () => {
    wrapper.find('a[href=""]').simulate('click', { preventDefault: () => {} });
    expect(mockGoBack).toHaveBeenCalled();
  });

  it('will check if the anchor has the alt tag for service manual tag', () => {
    wrapper.find('a[alt="Opens service manual"]').simulate('click');
    expect(wrapper.find('a[alt="Opens service manual"]')).toBeTruthy();
  });

  it('will check if the anchor has the alt tag for feedback form', () => {
    wrapper.find('a[alt="Opens feedback form"]').simulate('click');
    expect(wrapper.find('a[alt="Opens feedback form"]')).toBeTruthy();
  });

  it('should render PageTitle with the correct text', () => {
    expect(
      wrapper
        .find("PageTitle[title='Service improvement plan - GOV.UK']")
        .exists()
    ).toBeTruthy();
  });

  it('should render page with the correct heading', () => {
    expect(wrapper.find('h1').first().text()).toContain(
      "How we're improving the service"
    );
  });

  it('should render page with the correct links and text', () => {
    expect(
      wrapper
        .find('a[href="https://www.gov.uk/service-manual/service-standard"]')
        .first()
        .text()
    ).toEqual('Digital Service Standard (opens in new tab)');
    expect(wrapper.find('a[href="/some-feedback-url"]').first().text()).toEqual(
      'feedback (opens in new tab)'
    );
  });

  it('should render update information about how this service is being improved', () => {
    expect(wrapper.find('p').at(0).text()).toBe(
      'The Fish Exports Service helps businesses operating in the UK produce documents that enable fish exports to the EU and certain other countries.'
    );
    expect(wrapper.find('p').at(1).text()).toBe('The service:');
    expect(wrapper.find('p').at(2).text()).toBe(
      'As part of continuous improvement, the service aims to:'
    );
    expect(wrapper.find('p').at(3).text()).toBe(
      'Your feedback (opens in new tab) will help us to improve this service.'
    );

    expect(wrapper.find('li').at(0).text()).toBe(
      'uses your information in a secure way'
    );
    expect(wrapper.find('li').at(1).text()).toBe(
      'is working towards the WCAG 2.1 AA accessibility standard'
    );
    expect(wrapper.find('li').at(2).text()).toBe(
      'uses technology that allows it to be improved over time'
    );
    expect(wrapper.find('li').at(3).text()).toBe(
      'aims to meet the Government’s Digital Service Standard (opens in new tab).'
    );

    expect(wrapper.find('li').at(4).text()).toBe(
      'address and fix identified, WCAG 2.1 AA non-compliant accessibility issues (priority 1)'
    );
    expect(wrapper.find('li').at(5).text()).toBe(
      'carry out continuous research to make sure you can complete the service first time, and make improvements where necessary (priority 2)'
    );
    expect(wrapper.find('li').at(6).text()).toBe(
      'continue to research and test the service to ensure your needs are addressed (priority 3)'
    );
  });

  it('should display the correct title for catch certificates', () => {
    wrapper.find('ImproveServicePage').setState({
      backRoute: '/create-catch-certificate',
    });

    expect(wrapper.find('PageTitle').prop('title')).toBe(
      'Service improvement plan - Create a UK catch certificate - GOV.UK'
    );
  });

  it('should display the correct title for processing statements', () => {
    wrapper.find('ImproveServicePage').setState({
      backRoute: '/create-processing-statement',
    });

    expect(wrapper.find('PageTitle').prop('title')).toBe(
      'Service improvement plan - Create a UK processing statement - GOV.UK'
    );
  });

  it('should display the correct title for storage documents', () => {
    wrapper.find('ImproveServicePage').setState({
      backRoute: '/create-storage-document',
    });
    expect(wrapper.find('PageTitle').prop('title')).toBe(
      'Service improvement plan - Create a UK storage document - GOV.UK'
    );
  });

  it('should generate snapshot for the Plane Details page', () => {
    const { container } = render(wrapper);

    expect(container).toMatchSnapshot();
  });
});

describe('Improve service page in Welsh', () => {
  beforeEach(() => {
    jest.spyOn(document, 'getElementsByClassName').mockReturnValue([
      {
        title: '',
        href: '',
      },
    ]);

    wrapper = getWrapper('language', 'cy_UK');
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should load successfully', () => {
    expect(wrapper).toBeDefined();
  });

  it('should handle clicking back link', () => {
    wrapper.find('a[href=""]').simulate('click', { preventDefault: () => {} });
    expect(mockGoBack).toHaveBeenCalled();
  });

  it('will check if the anchor has the alt tag for service manual tag', () => {
    wrapper.find('a[alt="Opens service manual"]').simulate('click');
    expect(wrapper.find('a[alt="Opens service manual"]')).toBeTruthy();
  });

  it('will check if the anchor has the alt tag for feedback form', () => {
    wrapper.find('a[alt="Opens feedback form"]').simulate('click');
    expect(wrapper.find('a[alt="Opens feedback form"]')).toBeTruthy();
  });

  it('should render PageTitle with the correct text', () => {
    expect(
      wrapper
        .find("PageTitle[title='Cynllun gwella gwasanaeth - GOV.UK']")
        .exists()
    ).toBeTruthy();
  });

  it('should render page with the correct heading', () => {
    expect(wrapper.find('h1').first().text()).toContain(
      'Sut rydyn ni’n gwella’r gwasanaeth'
    );
  });

  it('should render page with the correct links and text', () => {
    expect(
      wrapper
        .find('a[href="https://www.gov.uk/service-manual/service-standard"]')
        .first()
        .text()
    ).toEqual(
      'Safon Gwasanaeth Digidol y Llywodraeth (yn agor mewn tab newydd)'
    );
    expect(wrapper.find('a[href="/some-feedback-url"]').first().text()).toEqual(
      'adborth (yn agor mewn tab newydd)'
    );
  });

  it('should render update information about how this service is being improved', () => {
    expect(wrapper.find('p').at(0).text()).toBe(
      'Mae’r Gwasanaeth Allforio Pysgod yn helpu busnesau sy’n gweithredu yn y DU i gynhyrchu dogfennau sy’n golygu bod modd allforio pysgod i’r UE ac i rai gwledydd eraill.'
    );
    expect(wrapper.find('p').at(1).text()).toBe('Mae’r gwasanaeth yn:');
    expect(wrapper.find('p').at(2).text()).toBe(
      'Er mwyn gwella’n barhaus, nod y gwasanaeth yw:'
    );
    expect(wrapper.find('p').at(3).text()).toBe(
      'Bydd eich adborth (yn agor mewn tab newydd) yn ein helpu i wella’r gwasanaeth hwn.'
    );

    expect(wrapper.find('li').at(0).text()).toBe(
      'defnyddio eich gwybodaeth mewn ffordd ddiogel'
    );
    expect(wrapper.find('li').at(1).text()).toBe(
      'gweithio tuag at safon hygyrchedd WCAG 2.1 AA'
    );
    expect(wrapper.find('li').at(2).text()).toBe(
      'defnyddio technoleg sy’n ei alluogi i wella dros amser'
    );
    expect(wrapper.find('li').at(3).text()).toBe(
      'Anelu at gyrraedd Safon Gwasanaeth Digidol y Llywodraeth (yn agor mewn tab newydd).'
    );

    expect(wrapper.find('li').at(4).text()).toBe(
      'canfod a datrys problemau hygyrchedd sydd ddim yn cydymffurfio â WCAG 2.1 AA (blaenoriaeth 1)'
    );
    expect(wrapper.find('li').at(5).text()).toBe(
      'gwneud gwaith ymchwil parhaus i wneud yn siŵr eich bod yn gallu cwblhau’r gwasanaeth y tro cyntaf, a gwneud gwelliannau lle bo angen (blaenoriaeth 2)'
    );
    expect(wrapper.find('li').at(6).text()).toBe(
      'parhau i ymchwilio a phrofi’r gwasanaeth fel ei fod yn cyflawni eich anghenion (blaenoriaeth 3)'
    );
  });

  it('should display the correct title for catch certificates', () => {
    wrapper.find('ImproveServicePage').setState({
      backRoute: '/create-catch-certificate',
    });

    expect(wrapper.find('PageTitle').prop('title')).toBe(
      'Cynllun gwella gwasanaeth - Create a UK catch certificate - GOV.UK'
    );
  });

  it('should display the correct title for processing statements', () => {
    wrapper.find('ImproveServicePage').setState({
      backRoute: '/create-processing-statement',
    });

    expect(wrapper.find('PageTitle').prop('title')).toBe(
      'Cynllun gwella gwasanaeth - Create a UK processing statement - GOV.UK'
    );
  });

  it('should display the correct title for storage documents', () => {
    wrapper.find('ImproveServicePage').setState({
      backRoute: '/create-storage-document',
    });
    expect(wrapper.find('PageTitle').prop('title')).toBe(
      'Cynllun gwella gwasanaeth - Create a UK storage document - GOV.UK'
    );
  });
});

describe('Improve service page - with no title', () => {
  beforeEach(() => {
    jest.spyOn(document, 'getElementsByClassName').mockReturnValue([undefined]);

    wrapper = getWrapper('language', 'en_UK');
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render', () => {
    expect(wrapper).toBeDefined();
  });
});

describe('Improve service page - with no previously selected language', () => {
  beforeEach(() => {
    jest.spyOn(document, 'getElementsByClassName').mockReturnValue([undefined]);

    wrapper = getWrapper();
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render', () => {
    expect(wrapper).toBeDefined();
  });
});

describe('Improve service page, loadData', () => {
  const store = { dispatch: jest.fn() };

  beforeEach(() => {
    getAllUserAttributes.mockReset();
    getAllUserAttributes.mockReturnValue({ type: 'GET_ALL_USER_ATTRIBUTES' });
  });

  it('should call the getAllUserAttributes to load the component', () => {
    ImproveServicePageComponent.loadData(store);

    expect(getAllUserAttributes).toHaveBeenCalledTimes(1);
    expect(getAllUserAttributes).toHaveBeenCalled();
  });
});