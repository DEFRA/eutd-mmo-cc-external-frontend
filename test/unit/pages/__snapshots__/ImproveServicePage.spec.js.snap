// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Improve service page in English should generate snapshot for the Plane Details page 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <a
        class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
        href=""
      >
        Back
      </a>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            How we're improving the service
          </h1>
        </div>
      </div>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <p>
            The 
            <strong>
              Fish Exports Service
            </strong>
             helps businesses operating in the UK produce documents that enable fish exports to the EU and certain other countries.
          </p>
          <p>
            The service:
          </p>
          <ul
            class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              style="font-size: 19px;"
            >
              uses your information in a secure way
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              style="font-size: 19px;"
            >
              is working towards the WCAG 2.1 AA accessibility standard
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              style="font-size: 19px;"
            >
              uses technology that allows it to be improved over time
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              style="font-size: 19px;"
            >
              aims to meet the Governmentâ€™s
               
              <a
                alt="Opens service manual"
                href="https://www.gov.uk/service-manual/service-standard"
                rel="noopener noreferrer"
                target="_blank"
              >
                Digital Service Standard (opens in new tab)
              </a>
              .
            </li>
          </ul>
          <br />
          <p>
            As part of continuous improvement, the service aims to:
          </p>
          <ul
            class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              style="font-size: 19px;"
            >
              address and fix identified, WCAG 2.1 AA non-compliant accessibility issues (priority 1)
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              style="font-size: 19px;"
            >
              carry out continuous research to make sure you can complete the service first time, and make improvements where necessary (priority 2)
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
              style="font-size: 19px;"
            >
              continue to research and test the service to ensure your needs are addressed (priority 3)
            </li>
          </ul>
          <br />
          <p>
            Your
             
            <a
              alt="Opens feedback form"
              href="/some-feedback-url"
              rel="noopener noreferrer"
              target="_blank"
            >
              feedback (opens in new tab)
            </a>
             
            will help us to improve this service.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
